compiler/src/test/kotlin/dev/zacsweers/metro/compiler/transformers/AssistedFactoryTransformerTest.kt

```diff
@@ -235,7 +235,28 @@ class AssistedFactoryTransformerTest : MetroCompilerTest() {
       expectedExitCode = COMPILATION_ERROR,
     ) {
       assertDiagnostics(
-        "e: ExampleClass.kt:6:7 `@AssistedFactory` target classes must have a single `@Inject`-annotated constructor or be annotated `@Inject` with only a primary constructor."
+        "e: ExampleClass.kt:10:7 `@AssistedFactory` target classes must have a single `@Inject`-annotated constructor or be annotated `@Inject` with only a primary constructor."
```

Comment:
Adjusted the report location from `ExampleClass` to `ExampleClassFactory.create`